### ISSUE TITLE
instrDescDebugInfo should store compile time handles

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#77789]
On xarch we currently call `emitDispCommentForHandle` with constants from the `instrDesc` in various places when printing jit disassembly. This function then treats these constants as various handles. This does not work for R2R where the constants and compile time handles are different.

We should probably just add a new field to `instrDescDebugInfo` (or use the existing `idMemCookie`, though currently its purpose is slightly different for some other handles) that stores the compile time handle. As a bonus this will work for arm platforms too, where the constants in the instructions can never be the full handle anyway.

**Comment by jakobbotsch** [comment]
This throws an `InvalidCastException` inside crossgen2, but the exception ends up being caught (via `runWithSPMIErrorTrap`, also unexpected) so we don't end up with user facing crash, we just get something like
```asm
488B0500000000       mov      rax, qword ptr [(reloc 0x40000000004200f0)]      ; hackishClassName
```

**Comment by jakobbotsch** [comment]
This was fixed by #78382
